### DISCUSSION TITLE
buf_proto_breaking_test: How to obtain an "image" for parameter "against_input"?

### DISCUSSION THREAD

**Discussion by sschnug** [rules-proto-grpc/rules_proto_grpc#150] [Q&A] [answered]
Hi,

after having taken some first (successfull) experiments with this tool, i wanted to have a look at `buf_proto_breaking_test`.

I wonder how it's meant to be used. 

The concept of some *image* was unclear to me, but [buf's doc](https://docs.buf.build/reference/images/) helped.

Now if i understand correctly, i need to prepare some image of my *last commit* (or better: cache them) and can then use `buf_proto_breaking_test` with my *current commit based input protos* against this image.

The strange thing, to me, it that i don't see any *functionality* within `rules_proto_grpc` to produce this image.

One probably could carefully follow *bufs docs* and create those, but i would be scared about being correct then as this tool here takes care of all those *protoc invocations* and generating images needed for this tasks also looks like one more *protoc invocation*.

**Shouldn't there be a rule / feature whose task is to produce a image giving some compatible input as `buf_proto_breaking_test` uses?**

Thanks for reading,
Sascha

**Comment by aaliddell** [comment, marked as answer]
You're right there's no way to create the images in this repo; I generally used the command at the [bottom of that page](https://docs.buf.build/reference/images#using-protoc-output-as-buf-input) as reference for testing IIRC. Creating a rule to automate that may be possible, since it's just calling protoc and we have that available. 

However, we wouldn't be able to have rules for 'compare to last commit' etc, since that is beyond the control of Bazel. The rule would have to be a `bazel run update_image` type rule that you then commit the output as your 'known good' image to compare to.

**Reply by aaliddell** [reply]
I have created an issue for this at #151
